.github/workflows/build.yml

```diff
@@ -0,0 +1,235 @@
+  build-tensorflow-macOS:
+    needs: tensorflow_opt-macOS
+    runs-on: macos-10.15
+    steps:
+      - run: true
+        if: needs.tensorflow_opt-macOS.outputs.status == 'found'
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          submodules: 'recursive'
+        if: needs.tensorflow_opt-macOS.outputs.status == 'missing'
+      - uses: ./.github/actions/select-xcode
+        with:
+          version: "12.1.1"
+        if: needs.tensorflow_opt-macOS.outputs.status == 'missing'
+      - uses: ./.github/actions/setup-tensorflow
+        if: needs.tensorflow_opt-macOS.outputs.status == 'missing'
+      - uses: ./.github/actions/build-tensorflow
+        with:
+          flavor: "--darwin-cpu"
+        if: needs.tensorflow_opt-macOS.outputs.status == 'missing'
+      - uses: ./.github/actions/package-tensorflow
+        if: needs.tensorflow_opt-macOS.outputs.status == 'missing'
+      - uses: actions/upload-artifact@v2
+        with:
+          name: ${{ needs.tensorflow_opt-macOS.outputs.cache_key }}
+          path: ${{ github.workspace }}/artifacts/home.tar.xz
+        if: needs.tensorflow_opt-macOS.outputs.status == 'missing'
+  build-lib_macOS:
+    runs-on: macos-10.15
+    needs: [ build-tensorflow-macOS, tensorflow_opt-macOS ]
+    strategy:
+      matrix:
+        build_flavor: ["tf", "tflite"]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/download-artifact@v2
+        with:
+          name: ${{ needs.tensorflow_opt-macOS.outputs.cache_key }}
+          path: ${{ github.workspace }}/
+        if: needs.tensorflow_opt-macOS.outputs.status == 'missing'
+      - uses: ./.github/actions/check_artifact_exists
+        with:
+          name: ${{ needs.tensorflow_opt-macOS.outputs.cache_key }}
+          path: ${{ github.workspace }}/
+          download: true
+        if: needs.tensorflow_opt-macOS.outputs.status == 'found'
+      - run: |
+          tar xkf ${{ github.workspace }}/home.tar.xz
+          rm ${{ github.workspace }}/home.tar.xz
+      - run: |
+          git status
+      - uses: ./.github/actions/select-xcode
+        with:
+          version: "12.1.1"
+      - uses: ./.github/actions/host-build
+        with:
+          flavor: ${{ matrix.build_flavor }}
+      - uses: ./.github/actions/package
+      - uses: actions/upload-artifact@v2
+        with:
+          name: "native_client.${{ matrix.build_flavor }}.tar.xz"
+          path: ${{ github.workspace }}/artifacts/native_client.tar.xz
+      - uses: actions/upload-artifact@v2
+        with:
+          name: "libdeepspeech.${{ matrix.build_flavor }}.zip"
+          path: ${{ github.workspace }}/artifacts/libdeepspeech.zip
+  build-python-macOS:
+    runs-on: macos-10.15
+    needs: [ build-lib_macOS, swig_macOS ]
+    strategy:
+      matrix:
+        build_flavor: ["tf", "tflite"]
+        python-version: [3.6.8, 3.7.9, 3.8.8, 3.9.2]
```

Comment:
nit (throughout the PR): GitHub style seems to be to use-dashes-in-names rather than underscores. In this PR they're mixed. Job-names use dashes, action_inputs use underscores, and in this matrix you have both. I think it's better to match GitHub's style and use-dashes-everywhere.

Reply:
Good catch. FTR for now there's a constraint in the `get_cache_key` job that I rely on `_` to separate between the job's name of a submodule and the platform, to make computation of the key smoother and with less parameters. I will have to ensure I properly document that in the action itself.